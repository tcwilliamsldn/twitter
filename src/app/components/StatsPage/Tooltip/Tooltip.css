.tooltip {
    background-color: #306aff;
    display: inline-block;
    padding: 12px 24px;
    text-align: center;
    margin-left: -50%;
    margin-top: 8px;

    &::after {
        position: absolute;
        height: 0;
        width: 0;
        bottom: calc(100% - 8px);
        left: 50%;
        border: solid transparent;
        content: '';
        pointer-events: none;
        border-bottom-color: #306aff;
        border-width: 8px;
        margin-left: calc(-50% - 8px);
    }
}

.date {
    color: #94b4fe;
}

.count {
    color: #ffffff;
    font-weight: bold;
}
