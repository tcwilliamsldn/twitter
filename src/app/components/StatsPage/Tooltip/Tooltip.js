import React from 'react';
import styles from './Tooltip.css';

const Tooltip = ({ retweet_count, date }) => {
    const formattedDate = new Date(date).toLocaleString('en-gb', {
        month: 'short',
        day: 'numeric'
    });

    return (
        <div className={styles.tooltip}>
            <div className={styles.date}>{formattedDate}</div>
            <div className={styles.count}>{`${retweet_count} retweets`}</div>
        </div>
    );
};

export default Tooltip;
