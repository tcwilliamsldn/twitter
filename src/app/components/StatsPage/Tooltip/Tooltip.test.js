import React from 'react';
import Tooltip from './Tooltip';
import { shallow } from 'enzyme';

describe('Tooltip', function() {
    const props = { date: 192837465, retweet_count: 'Tom' };

    it('renders a tooltip with correctly formatted date and retweet count', function() {
        const component = shallow(<Tooltip {...props} />);
        expect(component).toMatchSnapshot();
    });
});
