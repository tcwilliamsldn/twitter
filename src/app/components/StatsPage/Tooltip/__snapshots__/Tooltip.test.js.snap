// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip renders a tooltip with correctly formatted date and retweet count 1`] = `
<div
  className="tooltip"
>
  <div
    className="date"
  >
    Jan 3
  </div>
  <div
    className="count"
  >
    Tom retweets
  </div>
</div>
`;
